components/dashboard/src/prebuilds/list/PrebuildList.tsx

```diff
@@ -0,0 +1,211 @@
+            state: toApiStatus(filter.status),
+            ...(configurationFilter ? { configuration: { id: configurationFilter } } : {}),
+        },
+        sort: {
+            order: sortOrder === "desc" ? SortOrder.DESC : SortOrder.ASC,
+            field: sortBy,
+        },
+        pageSize: 30,
+    });
+
+    const prebuilds = useMemo(() => {
+        return data?.pages.map((page) => page.prebuilds).flat() ?? [];
+    }, [data?.pages]);
+
+    const hasMoreThanOnePage = (data?.pages.length ?? 0) > 1;
+
+    // This tracks any filters/search params applied
+    const hasFilters = !!searchTermDebounced || !!filter.status;
+
+    // Show the table once we're done loading and either have results, or have filters applied
+    const showTable = !isLoading && (prebuilds.length > 0 || hasFilters);
+
+    return (
+        <>
+            <div className="app-container mb-8">
+                <PageHeading title="Prebuilds" subtitle="Review prebuilds of your imported repositories." />
+
+                {isLoading && <LoadingState />}
+
+                {showTable && (
+                    <PrebuildsTable
+                        searchTerm={searchTerm}
+                        prebuilds={prebuilds}
+                        // we check isPreviousData too so we don't show spinner if it's a background refresh
+                        isSearching={isFetching && isPreviousData}
+                        isFetchingNextPage={isFetchingNextPage}
+                        hasNextPage={!!hasNextPage}
+                        filter={filter}
+                        sort={sort}
+                        hasMoreThanOnePage={hasMoreThanOnePage}
+                        onLoadNextPage={() => fetchNextPage()}
+                        onFilterChange={handleFilterChange}
+                        onSearchTermChange={setSearchTerm}
+                        onSort={handleSort}
+                    />
+                )}
+
+                {!showTable && !isLoading && <PrebuildListEmptyState />}
+                {isError && <PrebuildListErrorState error={error} />}
+            </div>
+        </>
+    );
+};
+
+const toApiStatus = (status: StatusOption): ListOrganizationPrebuildsRequest_Filter_State | undefined => {
+    switch (status) {
+        case "failed":
+            return ListOrganizationPrebuildsRequest_Filter_State.FAILED; // todo: adjust to needs of proper status
+        case "succeeded":
+            return ListOrganizationPrebuildsRequest_Filter_State.SUCCEEDED;
+        case "unfinished":
+            return ListOrganizationPrebuildsRequest_Filter_State.UNFINISHED;
+    }
+
+    return undefined;
+};
+
+const parseStatus = (params: URLSearchParams): StatusOption => {
+    const filter = params.get("prebuilds");
+    const validValues = Object.values(STATUS_FILTER_VALUES).filter((val) => !!val);
+    if (filter && validValues.includes(filter as any)) {
+        return filter as StatusOption;
+    }
+
+    return undefined;
+};
```

Comment:
Should be the same
```suggestion
const parseStatus = (params: URLSearchParams): StatusOption => {
    const filter = params.get("prebuilds");
    return STATUS_FILTER_VALUES.includes(filter as any);
};
```

Reply:
Wouldn't this return a `bool` instead of a `StatusOption`?

Reply:
Sorry, should be something like `return STATUS_FILTER_VALUES.includes(filter as any) && filter`

Changes is because of `Object.values` (it's already an array) and `filter`(will return undefined if not found) are not necessary